### PR TITLE
Update/to do's to users

### PR DESCRIPTION
**Pull request by flportilla**
Problem: 
All users had the rights to delete, post and create ToDos everywhere.

Actions taken:
[x] Update POST request handler in back end to allow only token validated users to create a new ToDo
[x] Update DELETE request handler in back end to allow only token validated users to delete ToDos
[x] Update GET request handler in back end to allow only token validated users to see the list of their own ToDos
[x] Update Todo and User schemas to reference each other in DB

Minor fixes:
[x] GET all users now shows id and list of Todos associated with the user
[x] Delete all .rest files.

### CODE REVIEW

File: api/controllers/todo.js [hunk 1]
```diff
@@ -8,21 +8,33 @@ const tokenExtractor = middleware.tokenExtractor
 
 //Handle GET all the todos
 todoRouter.get('/', tokenExtractor, userExtractor, async (request, response, next) => {
-  const allToDos = await Todo.find({})
-  response.json(allToDos)
-  next()
+
+  const allTodos = await Todo.find({})
```
Comment (yeion7) [inline api/controllers/todo.js:12]: Why not filter the user todos in the DB query instead?

File: api/controllers/todo.js [hunk 2]
```diff
@@ -42,8 +54,21 @@ todoRouter.put('/:id', tokenExtractor, userExtractor, async (request, response)
...
+  const todoInUser = user?.todos.filter(todo => todo.toString() === selectedTodo?.id)
+
+  if (todoInUser?.toString() === selectedTodo?.id) {
+    await Todo.findByIdAndRemove(request.params.id)
+
+    user.todos = user.todos.filter(todo => todo.toString() !== selectedTodo?.id)
+    await user.save()
+    response.status(204).end()
+  }
+
```
Comment (yeion7) [inline api/controllers/todo.js:70]: If you don't use a return here, the response in the line 73 alway will be called

File: api/controllers/todo.js (hunk 1 above)
Comment (flportilla) [reply api/controllers/todo.js:12]: https://mongoosejs.com/docs/tutorials/query_casting.html 
In mongoose you can find one document at the time, the other option is to install an external plugin (https://openbase.com/js/mongoose-find-and-filter/documentation) But the moment I install it everything crashes

File: api/controllers/todo.js (hunk 2 above)
Comment (flportilla) [reply api/controllers/todo.js]: Fixed

**Review by yeion7 — APPROVED** [review]

**Pull request merged**